lib/pengine/bundle.c

```diff
@@ -1487,6 +1502,40 @@ bundle_print_xml(pe_resource_t *rsc, const char *pre_text, long options,
     free(child_text);
 }
 
+static void
+bundle_output_xml(pe_resource_t *rsc, long options, pcmk__output_t * out)
+{
+    pe__bundle_variant_data_t *bundle_data = NULL;
+    char *child_text = NULL;
+    CRM_CHECK(rsc != NULL, return);
+
+    get_bundle_variant_data(bundle_data, rsc);
+
+    out->begin_list(out, "bundle", NULL, NULL);
+    out->set_str_prop(out, "id", rsc->id);
+    out->set_str_prop(out, "type", container_agent_str(bundle_data->agent_type));
+    out->set_str_prop(out, "image", bundle_data->image);
+    out->set_bool_prop(out, "unique", is_set(rsc->flags, pe_rsc_unique));
+    out->set_bool_prop(out, "managed", is_set(rsc->flags, pe_rsc_managed));
+    out->set_bool_prop(out, "failed", is_set(rsc->flags, pe_rsc_failed));
+
+    for (GList *gIter = bundle_data->replicas; gIter != NULL;
+         gIter = gIter->next) {
+        pe__bundle_replica_t *replica = gIter->data;
+
+        CRM_ASSERT(replica);
+        out->begin_list(out, "replica", NULL, NULL);
+        out->set_int_prop(out, "id", replica->offset);
```

Comment:
Unfortunately the current begin_list() interface doesn't have an argument for id (the second argument will be used as "name" though). It's an internal API though, so we could add a new argument.

Reply:
Should I add it as the last parameter, like
```
xml_begin_list(pcmk__output_t *out, const char *name,
               const char *singular_noun, const char *plural_noun, const char *id)
```

Reply:
Either last or after name would work